project/src/main/webapp/script.js

```diff
@@ -77,7 +77,9 @@ function showPage() {
 
 /** Retrieves list of topics and associated articles from the Backend Server in JSON form. */
 async function retrieveArticles(numArticles) {
-  const requestURL = '/news?num=' + numArticles;
+  const basePath = '/news';
+  const numParam = 'num=' + numArticles;
+  const requestURL = basePath + '?' + numParam;
```

Comment:
Can you put these at the top of the file? I could see you needing to hit different endpoints at some point going forward